lib/exp.js

```diff
@@ -0,0 +1,1206 @@
+  set: (bin, val, idx, policy = null, ctx = null) => [
+    ..._listModify(ctx, policy, lists.opcodes.SET, 2, 1),
+    ...idx,
+    ...val,
+    { op: exp.ops.CDT_LIST_CRMOD, listPolicy: policy },
+    ...bin
+  ],
+
+  clear: (bin, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.CLEAR, 0, 0),
+    ...bin
+  ],
+
+  sort: (bin, order, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.SORT, 1, 0),
+    ..._int(order),
+    ...bin
+  ],
+
+  removeAllByValue: (bin, val, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_ALL_BY_VALUE, 2, 0),
+    ..._int(lists.RETURN_NONE),
+    ...val,
+    ...bin
+  ],
+
+  removeByValueList: (bin, values, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_VALUE_LIST, 2, 0),
+    ..._int(lists.RETURN_NONE),
+    ...values,
+    ...bin
+  ],
+
+  removeByValueRange: (bin, end, begin, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_VALUE_INTERVAL, 3, 0),
+    ..._int(lists.RETURN_NONE),
+    ...begin,
+    ...end,
+    ...bin
+  ],
+
+  removeByRelRankRangeToEnd: (bin, rank, val, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_VALUE_REL_RANK_RANGE, 3, 0),
+    ..._int(lists.RETURN_NONE),
+    ...val,
+    ...rank,
+    ...bin
+  ],
+
+  removeByRelRankRange: (bin, count, rank, val, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_VALUE_REL_RANK_RANGE, 4, 0),
+    ..._int(lists.RETURN_NONE),
+    ...val,
+    ...rank,
+    ...count,
+    ...bin
+  ],
+
+  removeByIndex: (bin, idx, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_INDEX, 2, 0),
+    ..._int(lists.RETURN_NONE),
+    ...idx,
+    ...bin
+  ],
+
+  removeByIndexRangeToEnd: (bin, idx, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_INDEX_RANGE, 2, 0),
+    ..._int(lists.RETURN_NONE),
+    ...idx,
+    ...bin
+  ],
+
+  removeByIndexRange: (bin, count, idx, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_INDEX_RANGE, 3, 0),
+    ..._int(lists.RETURN_NONE),
+    ...idx,
+    ...count,
+    ...bin
+  ],
+
+  removeByRank: (bin, rank, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_RANK, 2, 0),
+    ..._int(lists.RETURN_NONE),
+    ...rank,
+    ...bin
+  ],
+
+  removeByRankRangeToEnd: (bin, rank, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_RANK_RANGE, 2, 0),
+    ..._int(lists.RETURN_NONE),
+    ...rank,
+    ...bin
+  ],
+
+  removeByRankRange: (bin, count, rank, ctx = null) => [
+    ..._listModify(ctx, null, lists.opcodes.REMOVE_BY_RANK_RANGE, 3, 0),
+    ..._int(lists.RETURN_NONE),
+    ...rank,
+    ...count,
+    ...bin
+  ]
+}
+
+/*********************************************************************************
+ * MAP MODIFY EXPRESSIONS
+ *********************************************************************************/
+
+function getMapType (type, returnType, isMulti) {
+  let expected = type
```

Comment:
Need to add more unit tests using different map types/return types, to ensure this function gets covered.